### ISSUE TITLE
Start KWallet from ownCloud? KDE integration 'no keychain service available'

### ISSUE THREAD

**Issue by sirio81** [owncloud/client#4601]
### Expected behaviour
owncloud client should start without asking the already saved password

### Actual behaviour
a windows is opened asking to insert the password with the message 'no keychain service available'

### Steps to reproduce
1. configure owncloud client
2. restart kde

### Client configuration
Client version: 2.1.1

Operating system: debian 8.3 64 bit.

OS language: italian

Installation path of client: /usr/bin/owncloud

### Logs

1. Output of `owncloud --logwindow`

03-23 11:18:49:931 0x24511e0 Checking server and authentication 
03-23 11:18:49:933 0x24511e0 Trying to look up system proxy 
03-23 11:18:49:933 0x26da0c0 virtual void OCC::SystemProxyRunnable::run() Starting system proxy lookup 
03-23 11:18:50:119 0x24511e0 No system proxy set by OS 
03-23 11:18:50:120 0x24511e0 Object::connect: No such signal QNetworkReplyImpl::encrypted()
03-23 11:18:50:120 0x24511e0 !!! OCC::CheckServerJob created for "http://demo8.extensys.it" + "status.php" "OCC::ConnectionValidator" 
03-23 11:18:50:371 0x24511e0 status.php returns:  QMap(("edition", QVariant(QString, "") ) ( "installed" ,  QVariant(bool, true) ) ( "maintenance" ,  QVariant(bool, false) ) ( "version" ,  QVariant(QString, "8.2.2.2") ) ( "versionstring" ,  QVariant(QString, "8.2.2") ) )    0  Reply:  QNetworkReplyImpl(0x29e9ea0) 
03-23 11:18:50:372 0x24511e0 ** Application: ownCloud found:   QUrl( "http://demo8.extensys.it/status.php" )   with version  "8.2.2" ( "8.2.2.2" ) 
03-23 11:18:50:373 0x24511e0 AccountState connection status change:  "Undefined" -> "Credentials Wrong" 
03-23 11:18:50:373 0x24511e0 AccountState state change:  "Disconnesso" -> "Errore di configurazione" 
03-23 11:18:50:374 0x24511e0 Account "paola@demo8.extensys.it" disconnected, terminating or descheduling sync folders 
03-23 11:18:53:048 0x24511e0 virtual void OCC::PassiveUpdateNotifier::backgroundCheckForUpdate() 
03-23 11:18:53:049 0x24511e0 virtual void OCC::OCUpdater::backgroundCheckForUpdate() checking for available update 
03-23 11:18:53:666 0x24511e0 Client is on latest version! 
03-23 11:18:58:886 0x24511e0 AccountState state change:  "Errore di configurazione" -> "Disconnesso"
<here I press cancel>
03-23 11:18:58:887 0x24511e0 Account "paola@demo8.extensys.it" disconnected, terminating or descheduling sync folders

### Notes:

I checked that the password save in kde wallet manager is the right one.
If I press cancel, close and re-open owncloud client, it ask for the password again.
If I insert the password, close and re-open owncloud client, it doesn't ask for password again.

**Comment by sirio81** [comment]
I can add a detail.
I disable owncloud to start at boot (kde login login).
If I open the wallet manger and open kde wallet, the owncloud starts without asking for password.
So it's evident that it's failing or not trying to open kde wallet.
My wallet is not protected by password.

**Comment by guruz** [comment]
Similar: https://github.com/owncloud/client/issues/3896 https://github.com/owncloud/client/issues/1590 https://github.com/owncloud/client/issues/3896

**Comment by sirio81** [comment]
it seems like an old problem that nobody is taking care of.

**Comment by sirio81** [comment]
At least I've been able to find a work around. It's simply matter of starting kwalletd before starting owncloud. 

```
system settings / Startup and shutdown / Desktop File / ownCloud / Properties... / Application
/usr/bin/kwalletd 2> /dev/null; /usr/bin/owncloud
```

**Comment by guruz** [comment]
@ogoffart Why does KDE not always start KWalletd?

**Comment by sirio81** [comment]
I have no idea, but I see the other applications such as krdc do not have problem starting it.
I'm going to ask anyway in the devel kde mailing list if it should start along with kde.
If it doesn't cost to much pain, it may worth to let owncloud start it if not already running.
I'll report back as soon as I get an answer.

**Comment by dragotin** [comment]
I think the qtkeychain lib that we're using simply does not have the feature to start kwalletd.

**Comment by sirio81** [comment]
From kde-devel:

> kwalletd is started on-demand through dbus activation [1] when one
> tries to use the org.kde.kwalletd(5) bus name. If it is not started
> dbus is not set up correctly or the service files are missing or
> qtkeychain manages to somehow not trigger dbus activation.
>  https://www.freedesktop.org/wiki/IntroductionToDBus/#activation

I guess the issue is the last option.
Are you in direct contact with qtkeychain developers?
Could you address this issue to somebody?

**Comment by dragotin** [comment]
@sirio81 thanks, good information. 

Yes, upstream of qtkeychain is here: https://github.com/frankosterfeld/qtkeychain 
You could create an issue there, but probably a patch would be even cooler...

**Comment by sirio81** [comment]
2016-03-24 10:55 GMT+01:00 Klaas Freitag <notifications@github.com>:

> Yes, upstream of qtkeychain is here:
> https://github.com/frankosterfeld/qtkeychain
> You could create an issue there, but probably a patch would be even
> cooler...
>
I'm not a developer but I'm going to report the issue.

**Comment by sirio81** [comment]
Let's keep an eye on: https://github.com/frankosterfeld/qtkeychain/issues/67

**Comment by guruz** [comment]
Some people worked around this: http://the-technoholik.blogspot.de/2015/01/owncloud-client-asking-for-password.html

**Comment by sirio81** [comment]
I also found a workaround (it's kind of the same) as written in my [previous post](https://github.com/owncloud/client/issues/4601#issuecomment-200377228), but it's still a workaround and **not a solution**.
I contacted also the developer and he says he's not motivated to actively support the project.
I'm trying to find someone who's able create a patch to add kwalletd support to libqtkeychain.

**Comment by sirio81** [comment]
Could you please update qtkeychain debian package to latest version?
Form what the author says and reading also this [post](https://github.com/owncloud/client/issues/4629) it seems the bug is fixed in the latest release.

**Comment by guruz** [comment]
@dragotin @jnweiger Added this to milestone as a heads-up for our linux packages

**Comment by dragotin** [comment]
I think @danimo already updated. Is that right, @danimo ?

**Comment by danimo** [comment]
The nightlies come with a recent version of QtKeyChain (0.6.1), see https://build.opensuse.org/package/show/isv:ownCloud:community:nightly/qtkeychain and https://build.opensuse.org/package/show/isv:ownCloud:community:nightly/qt5keychain

**Comment by guruz** [comment]
@sirio81 Could you help testing this?

**Comment by rperezb** [comment]
Close for inactivity, @sirio81 if this were not closed within the latest version, 2.2.0, please reopen

**Comment by sirio81** [comment]
Sorry, I've been busy with other tasks.
I installed owncloud from the nightly

```
apt-show-versions owncloud-client 
owncloud-client:amd64/unknown 2.2.0~nightly2016512-1.1 uptodate
```

I notice that it didn't install also libqtkeychain0 (I guess a dependency issue).
I install it

```
apt-show-versions libqtkeychain0
libqtkeychain0:amd64/unknown 0.6.2-11.1 uptodate
```

but I still get 'No key chain srevice available'.

**Comment by Paviluf** [comment]
I can confirm the "No key chain service available" problem too on Kubuntu 14.04.

**Comment by ogoffart** [comment]
This is a problem with qtkeychain. As this is not a bug in owncloud client, i'm closing this issue. It can still be tracked in https://github.com/frankosterfeld/qtkeychain/issues/67
